### PR TITLE
MikroTik RouterOS 'ip neighbor print detail' template

### PR DESCRIPTION
**Pull request by PavloSkliarenko**
Template + test

### CODE REVIEW

File: ntc_templates/templates/mikrotik_routeros_ip_neighbor_print_detail.textfsm [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Value NUM (\d+)
+Value INTERFACE (\S+)
+Value ADDRESS (\S+)
```
Comment (jvanderaa) [inline ntc_templates/templates/mikrotik_routeros_ip_neighbor_print_detail.textfsm:3]:
```suggestion
Value IP_ADDRESS (\S+)
```

File: ntc_templates/templates/mikrotik_routeros_ip_neighbor_print_detail.textfsm [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Value ADDRESS4 (\S+)
```
Comment (jvanderaa) [inline ntc_templates/templates/mikrotik_routeros_ip_neighbor_print_detail.textfsm:4]: What is the point of this versus ADDRESS?

**Review by jvanderaa — DISMISSED** [review]

File: ntc_templates/templates/mikrotik_routeros_ip_neighbor_print_detail.textfsm (hunk 2 above)
Comment (PavloSkliarenko) [reply ntc_templates/templates/mikrotik_routeros_ip_neighbor_print_detail.textfsm]: I couldn't find description of this value on the official [documentation](https://help.mikrotik.com/docs/display/ROS/Neighbor+discovery), but it has `address6` value description. So I assume that MikroTik has separate value for IPv4 address also.
Renamed this value to IPV4_ADDRESS

File: ntc_templates/templates/mikrotik_routeros_ip_neighbor_print_detail.textfsm [hunk 3]
```diff
@@ -0,0 +1,23 @@
+Value NUM (\d+)
```
Comment (jmcgill298) [inline ntc_templates/templates/mikrotik_routeros_ip_neighbor_print_detail.textfsm:1]:
```suggestion
Value NUMBER (\d+)
```

File: ntc_templates/templates/mikrotik_routeros_ip_neighbor_print_detail.textfsm [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+Value SOFTWARE_ID (\S+)
+Value BOARD (\S+)
+Value IPV6 (yes|no)
+Value INTERFACE_NAME (.*?(?="))
+Value SYSTEM_DESCRIPTION (.*?(?="))
+Value SYSTEM_CAPS ([a-zA-Z0-9,-]*)
+Value SYSTEM_CAPS_ENABLED ([a-zA-Z0-9,-]*)
+
+Start
+  ^\s*${NUM}\s+interface=${INTERFACE}(?:\s+address=${IP_ADDRESS})?(?:\s+address4=${IPV4_ADDRESS})?\s+mac-address=${MAC_ADDRESS}\s+identity="${IDENTITY}"\s+platform="${PLATFORM}"\s+version="${VERSION}"\s+unpack=${UNPACK}\s+age=${AGE}(?:\s+uptime=${UPTIME})?(?:\s+software-id="${SOFTWARE_ID}")?(?:\s+board="${BOARD}")?(?:\s+ipv6=${IPV6})?\s+interface-name="${INTERFACE_NAME}"(?:\s+system-description="${SYSTEM_DESCRIPTION}")?\s+system-caps="?${SYSTEM_CAPS}"?\s+system-caps-enabled="?${SYSTEM_CAPS_ENABLED}"?\s*$$ -> Record
```
Comment (jmcgill298) [inline ntc_templates/templates/mikrotik_routeros_ip_neighbor_print_detail.textfsm:21]:
```suggestion
  ^\s*${NUMBER}\s+interface=${INTERFACE}(?:\s+address=${IP_ADDRESS})?(?:\s+address4=${IPV4_ADDRESS})?\s+mac-address=${MAC_ADDRESS}\s+identity="${IDENTITY}"\s+platform="${PLATFORM}"\s+version="${VERSION}"\s+unpack=${UNPACK}\s+age=${AGE}(?:\s+uptime=${UPTIME})?(?:\s+software-id="${SOFTWARE_ID}")?(?:\s+board="${BOARD}")?(?:\s+ipv6=${IPV6})?\s+interface-name="${INTERFACE_NAME}"(?:\s+system-description="${SYSTEM_DESCRIPTION}")?\s+system-caps="?${SYSTEM_CAPS}"?\s+system-caps-enabled="?${SYSTEM_CAPS_ENABLED}"?\s*$$ -> Record
```

File: tests/mikrotik_routeros/ip_neighbor_print_detail/mikrotik_routeros_ip_neighbor_print_detail.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
+parsed_sample:
+  - num: "0"
```
Comment (jmcgill298) [inline tests/mikrotik_routeros/ip_neighbor_print_detail/mikrotik_routeros_ip_neighbor_print_detail.yml:3]:
```suggestion
  - number: "0"
```

**Review by jmcgill298 — APPROVED** [review]

File: ntc_templates/templates/index [hunk 6]
```diff
@@ -707,6 +707,7 @@ mikrotik_routeros_log_print_detail_without-paging.textfsm, .*, mikrotik_routeros
 mikrotik_routeros_routing_bgp_peer_print_status.textfsm, .*, mikrotik_routeros, [[/]]r[[outing]] bg[[p]] p[[eer]] p[[rint]] s[[tatus]]
 mikrotik_routeros_ip_arp_print_without-paging.textfsm, .*, mikrotik_routeros, [[/]]ip arp print without-paging
 mikrotik_routeros_ip_address_export_verbose.textfsm, .*, mikrotik_routeros, [[/]]ip a[[ddress]] e[[xport]] [[verbose]]
+mikrotik_routeros_ip_neighbor_print_detail.textfsm, .*, mikrotik_routeros, [[/]]ip n[[eighbor]] p[[rint]] d[[etail]]
 mikrotik_routeros_interface_ethernet_print.textfsm, .*, mikrotik_routeros, [[/]]in[[terface]] et[[hernet]] pr[[int]]
```
Comment (jmcgill298) [inline ntc_templates/templates/index:711]:
```suggestion
mikrotik_routeros_interface_ethernet_print.textfsm, .*, mikrotik_routeros, [[/]]in[[terface]] et[[hernet]] pr[[int]]
mikrotik_routeros_ip_neighbor_print_detail.textfsm, .*, mikrotik_routeros, [[/]]ip n[[eighbor]] p[[rint]] d[[etail]]
```

**Pull request merged**